[Corey Siah Jia Rong] iP

CoreySiah opened the pull request:
# NAT
Your all-in one chatbot to free your mind.

**Capabilities:**
- *Add tasks*
- ~~Delete tasks~~
- <sub>Mark/unmark tasks</sub>
- <ins>List</ins>
- <sup>Find</sup>

**Top UFC fighters:**
1. Average encik's grandma
2. Me

**Test it yourself!**
```
function getABanana() {
  System.out.println("Here's a banana!");
}
```

**Task list:**
- [ ] Get a super computer
- [ ] Steal the moon
- [ ] Get vector'ed

🥰😩🤚🏻

[!TIP]
> Read the code at your own risk!

Not a rick roll
[https://www.youtube.com/watch?v=dQw4w9WgXcQ](url)

This site was built using [GitHub Pages](https://pages.github.com/).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

CoreySiah committed:
Committed Duke Level-0: Rename, Greet, Exit

CoreySiah committed:
Committed Duke Level-1: Echo

CoreySiah committed:
Committed Duke Level-2: Add, List

CoreySiah committed:
Committed Duke Level-3: Mark as Done

CoreySiah committed:
Halfway Level 4

CoreySiah committed:
75% Level:4

CoreySiah committed:
Completed Duke Level-4: ToDo, Event, Deadline

CoreySiah committed:
Completed Duke Level-5: Handle Errors

CoreySiah committed:
Add Duke Level-6: Delete

CoreySiah committed:
Update

CoreySiah committed:
Update

CoreySiah committed:
Add 50% of Duke Level-7: Save

CoreySiah committed:
Add Level-7

CoreySiah committed:
Add Level-8

CoreySiah committed:
Add Level-8

CoreySiah committed:
Add OOP classes

dexterkwxn commented on src/main/java/Nat.java:
```diff
@@ -0,0 +1,299 @@
+import java.util.Scanner;
+
+import java.util.ArrayList;
+
```
Maybe consider removing redundant lines between imports?

dexterkwxn commented on src/main/java/Nat.java:
```diff
@@ -0,0 +1,299 @@
+                default:
+                    System.out.println(SPACER + " Oh noo! I don't know what you mean; please try again!\n" + HORIZONTAL_LINE);
+                    break;
+            }
```
The cases should not be indented according to the coding standard guidelines. 
Refer to https://se-education.org/guides/conventions/java/intermediate.html#layout for more information.

dexterkwxn commented on src/main/java/Nat.java:
```diff
@@ -0,0 +1,299 @@
+                        break;
+                    default:
+                        continue;
+                }
```
Similar issue here with indentation.

dexterkwxn reviewed (commented):
Some small nits with indentation and redundant lines, otherwise good job!

CoreySiah committed:
Add Duke A-MoreOOP: Use More OOP

CoreySiah committed:
Add Duke A-Packages: Organize into Packages

CoreySiah committed:
Merge remote-tracking branch 'origin/add-gradle-support'

CoreySiah committed:
Add Duke A-Gradle: Use Gradle

CoreySiah committed:
Add Duke A-JUnit: Add JUnit Tests

CoreySiah committed:
Fix Tags (Tags weren't getting logged)

CoreySiah committed:
Add Duke A-JavaDoc: JavaDoc

CoreySiah committed:
Add Duke A-CodingStandard: Follow the Coding Standard

CoreySiah committed:
Add Duke Level-9: Find

CoreySiah committed:
Fix branch-level-8 tag

CoreySiah committed:
Add Duke A-CheckStyle: Use CheckStyle

CoreySiah committed:
Add Duke Level-10: GUI

yuto1115 commented on src/main/java/Nat/Deadline.class:
Java class file should not be tracked by git. (also applied to all other class files)

I suggest to modify your .gitignore file.

yuto1115 commented on src/main/java/Nat/Nat.java:
```diff
@@ -0,0 +1,96 @@
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+    private Scanner scanner;
```
Do you think that it is better to mark these members as `final`?

yuto1115 commented on src/main/java/Nat/Task.java:
```diff
@@ -0,0 +1,74 @@
+ * The Task class represents the class for general tasks.
+ */
+public class Task {
+    protected String taskName;
+    protected boolean isDone;
```
It seems that the child classes of `Task` are not accessing to these members.

Maybe better to use `private` instead of `protected`?

yuto1115 reviewed (commented):
Overall, your code is well-organised and looks good to me. Just some nits to follow OOP principles more.

CoreySiah committed:
Add Duke Level-10: GUI part 2

CoreySiah committed:
Add Nat.java GUI

CoreySiah committed:
Fix checkstyle errors and add necessary JavaDocs

There are several classes that failed the checkstyle tests. Some
classes are missing Javadocs.

Fixing all those checkstyle errors and adding JavaDocs comments
improves the code quality.

As a step towards writing clean readable code, let's keep in mind
standard Java and Git practices when writing code. Doing so will
make the subsequent fixes easier.

CoreySiah committed:
Add Duke A-Assertions: Use Assertions

CoreySiah committed:
Revert "Add Duke A-Assertions: Use Assertions"

This reverts commit 1f720bb4266b575ddf5210ee0df3a563694ab635.

CoreySiah committed:
Add Duke A-Assertions: Use Assertions

CoreySiah committed:
Add Duke A-CodeQuality

CoreySiah committed:
Add Duke A-Streams: Use Streams

CoreySiah committed:
Merge branch 'branch-A-CodeQuality' into branch-A-Assertions

CoreySiah committed:
Merge branch-A-CodeQuality into Master branch

CoreySiah committed:
Merge branch 'branch-A-Streams'

CoreySiah committed:
Merge pull request #3 from CoreySiah/branch-A-Assertions

Merge branch-A-Assertions into branch-A-CodeQuality

CoreySiah committed:
Merge branch 'master' into branch-A-Assertions

CoreySiah committed:
Merge pull request #2 from CoreySiah/branch-A-Assertions

Merge branch-A-Assertions branch into master branch

CoreySiah committed:
Add BCD-Extension

CoreySiah committed:
Add Duke A-BetterGui: Better GUI

CoreySiah committed:
Add Duke A-Personality: A unique personality

CoreySiah committed:
Add Duke A-MoreTesting: More testing

CoreySiah committed:
Add Duke A-AiAssisted: Enhance using AI tools

CoreySiah committed:
Add AI.md

CoreySiah committed:
Add Ui.png and minor changes to TaskList.java

CoreySiah committed:
Add Duke A-UserGuide: User Guide

CoreySiah committed:
Add README.md

CoreySiah committed:
Fix content table formatting

CoreySiah committed:
Fix command summary

CoreySiah committed:
Fix spacing issues

CoreySiah committed:
Add content table

CoreySiah committed:
Add gradle configurations

CoreySiah committed:
Merge branch 'branch-A-CodeQuality'

CoreySiah committed:
Edit build.gradle
